lxc/exec.go

```diff
@@ -214,6 +237,6 @@ func (c *execCmd) run(conf *config.Config, args []string) error {
 		termios.Restore(cfd, oldttystate)
 	}
 
-	os.Exit(ret)
-	return fmt.Errorf(i18n.G("unreachable return reached"))
+	os.Exit(int(op.Metadata["return"].(float64)))
```

Comment:
Yeah, that's what JSON ints are unpacked as, trying to parse as anything else causes a panic.